editor/src/components/shared/scoped-variables.ts

```diff
@@ -83,26 +99,79 @@ function getVariablesFromComponent(
   const jsxComponentVariables = variablesInScopeFromEditorState[elementPathString] ?? {}
   return {
     filePath: jsxComponent?.name ?? 'Component',
-    variables: Object.entries(jsxComponentVariables).flatMap(([name, value]) => {
-      const type = getTypeByValue(value)
-      const variable = {
-        name,
-        value,
-        type,
+    variables: generateVariableTypes(jsxComponentVariables),
+  }
+}
+
+function getComponentPropsInScope(
+  topLevelElements: TopLevelElement[],
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): AllVariablesInScope {
+  const jsxComponent = topLevelElements.find((el) => isUtopiaJSXComponent(el)) as UtopiaJSXComponent
+  const jsxComponentPropNamesUsed = jsxComponent?.propsUsed ?? []
+
+  const jsxComponentProps = omitWithPredicate(
+    getContainerPropsValue(elementPath, jsxMetadata),
+    (key) => !jsxComponentPropNamesUsed.includes(key as string),
```

Comment:
Do we know this value is for sure a `string`? If not I'd add a check for that to the predicate.